subsys/storage/stream/stream_flash.c

```diff
@@ -144,6 +146,25 @@ int stream_flash_buffered_write(struct stream_flash_ctx *ctx, const u8_t *data,
 	}
 
 	if (flush && ctx->buf_bytes > 0) {
+		addenum = ctx->buf_bytes %
+				 flash_get_write_block_size(ctx->fdev);
+		if (addenum) {
+			/*
+			 * take advantage fact that non written memory should
+			 * be erased in order to get now the erased byte-value.
+			 */
+			rc = flash_read(ctx->fdev,
+					ctx->offset + ctx->bytes_written,
+					(void *)&filler,
+					1);
+
+			if (rc != 0) {
+				return rc;
+			}
+			memset(ctx->buf + ctx->buf_bytes, filler, addenum);
```

Comment:
This is at risk for writing past the end of `ctx->buf`.  That should be avoided by adding a check in `stream_flash_init()` that `buf_len` is a multiple of the write block size.

Reply:
agree - was thought about such check before, but thought it is overkill